content/blog/how-to-use-nodejs-loader.en.mdx

```diff
@@ -0,0 +1,289 @@
+---
+title: How to use Node.js ESM loader
+description: With this article, you will understand Node.js loaders and how to use them.
+date: 2024-11-08
+authors: AugustinMauroy
+---
+
+### What is a loader?
+
+A loader in the context of Node.js is a mechanism that allows developers to customize the behavior of module resolution and loading. It provides hooks that can be used to intercept and modify how Node.js handles `import` statements and `require` calls. This customization can be particularly useful for tasks such as:
+
+- Loading non-JavaScript files (e.g., TypeScript, CoffeeScript)
+- Transpiling code on-the-fly
+- Implementing custom module formats
+- Overriding or extending the default module resolution logic
+
+By using loaders, developers can tailor the module system to better suit their specific needs, making it easier to work with different file types and module formats. This can be especially beneficial during development and testing phases, where flexibility and experimentation are key.
+
+### What a loader contains?
+
+A loader in Node.js is essentially a set of custom functions that allow you to control how modules are resolved and loaded. These functions, known as hooks, provide a way to intercept and modify the default behavior of Node.js when it processes `import` statements and `require` calls. The two main hooks that a loader contains are the `resolve` hook and the `load` hook.
+
+#### The `resolve` Hook
+
+The `resolve` hook is responsible for determining the location of a module given a specifier. A specifier is the string or URL that you pass to `import` or `require`. For example, when you write `import 'some-module'`, the `resolve` hook helps Node.js figure out where `some-module` is located.
```

Comment:
```suggestion
The `resolve` hook is responsible for determining the location of a module, given a specifier. A specifier is the string or URL that you pass to `import` or `require`. For example, when you write `import 'some-module'`, the `resolve` hook helps Node.js figure out where `some-module` is located.
```

technically, the thing you passed to `require()` is called an identifier (or "id").

Reply:
Humm yes but most of user will use and understand it as a "module" so I will take this as it